### ISSUE TITLE
Bug: Conflict with eslint-plugin-yml

### ISSUE THREAD

**Issue by paul4156** [typescript-eslint/typescript-eslint#11332]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

I expected `typescriptEslint.configs.recommendedTypeChecked` to work with [eslint-plugin-yml](https://github.com/ota-meshi/eslint-plugin-yml).

But eslint parses yaml files as ts files and reports false positives:

```
/path/to/lint-yaml/config.yaml
  2:12  error  Parsing error: ';' expected
```

I can only use a workaround as in https://github.com/ota-meshi/eslint-plugin-yml/issues/261. Is there a better way? Thanks.

### Reproduction Repository Link

https://github.com/paul4156/lint-yaml

### Repro Steps

1) clone the repo
2) `npm install`
3) `npm run lint`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.34.1` |
| `TypeScript`                           | `5.8.3` |
| `ESLint`                               | `9.29.0` |
| `node`                                 | `v22.15.1` |
| `eslint-plugin-yml`            | `1.18.0` |

**Comment by bradzacher** [comment]
It is up to the user to configure plugins and rules so they run on supported files only.

You need to leverage `files` and `extends` to scope our configs to just TS files.

https://eslint.org/docs/latest/use/configure/configuration-files#specifying-files-and-ignores
